modules/core/04-channel/types/errors.go

```diff
@@ -45,4 +45,8 @@ var (
 	ErrInvalidUpgradeSequence          = errorsmod.Register(SubModuleName, 29, "invalid upgrade sequence")
 	ErrUpgradeNotFound                 = errorsmod.Register(SubModuleName, 30, "upgrade not found")
 	ErrIncompatibleCounterpartyUpgrade = errorsmod.Register(SubModuleName, 31, "incompatible counterparty upgrade")
+	ErrInvalidUpgradeError             = errorsmod.Register(SubModuleName, 32, "invalid upgrade error")
+	ErrUpgradeRestoreFailed            = errorsmod.Register(SubModuleName, 33, "restore failed")
```

Comment:
this error is coming from the [WriteUpgradeTimeoutPR](https://github.com/cosmos/ibc-go/pull/3797/files#diff-0fca3dc84b5a25a4a314bceed2fdee673da45a5381aed16d164bb84cd891642e)